### ISSUE TITLE
Fix Request: false positive in AvoidInfiniteRecursion

### ISSUE THREAD

**Issue by jborgers** [jborgers/PMD-jPinpoint-rules#626]
The rule seems confused with nested classes:

```
public final class Conditions {
    public class ConditionBuilder {
       private final Collection<Condition> conditions = new ArrayList<>();
   
       public ConditionBuilder isTrue(Supplier<Boolean> isTrue, String name) {
           conditions.add(Conditions.this.isTrue(isTrue, name));  // <-- false positive 
           return this;
       }

   public Condition isTrue(Supplier<Boolean> isTrue, String name) {
     return new Condition() {
       @Override
       public String description() {
          return name + " == true";
       }
    };
  }
}
```
